### ISSUE TITLE
StorageUse leads to multiple entries for one event

### ISSUE THREAD

**Issue by tobias74** [Microsoft/ApplicationInsights-JS#796]
I have found that having "isStorageUseDisabled" set to false will cause some events to be sent multiple times. Well I am not actually sure that they get sent multiple times, but they appear multiple times in the protocols on the azure-portal.

Setting "isStorageUseDisabled" to true resolved the issue for me.

Here is my according question on SO:

[https://stackoverflow.com/questions/54929563/azure-application-insights-shows-the-same-event-at-identical-timestamps-multiple/54948285#54948285](https://stackoverflow.com/questions/54929563/azure-application-insights-shows-the-same-event-at-identical-timestamps-multiple/54948285#54948285)

I have found another issue here: [https://github.com/Microsoft/ApplicationInsights-JS/issues/436](https://github.com/Microsoft/ApplicationInsights-JS/issues/436) This might actually be the same thing?

**Question:** When are the events removed from the storage? Only after a call to the tracking-server was successfully completed? I observed that when I reload the page while the tracking-request is running, the events are not removed from the storage. This could explain why sometimes events are duplicated?

**(I am using version 1.0.20)**

**Comment by jpiyali** [comment]
@tobias74  - Items are removed from storage from after successful send. Issue can occur sometimes if you have multiple tabs open. Do you have multiple tabs open?


https://www.npmjs.com/package/@microsoft/applicationinsights-web/v/2.0.0-beta.3 
Beta SDK has the ability to provide prefix for localstorage/cookie name. Setting: 
{
instrumentationKey: "xxxxxxx"
**namePrefix: "App1"**

}


If you provide this setting, local storage will be unique between multiple sessions and not cause any duplicates.

**Comment by jpiyali** [comment]
Please reopen if you still see issues in beta SDK using namePrefix setting or if multiple tabs are not opened.

**Comment by tobias74** [comment]
For me the issue occurred in case the answer from the insights-server took too long. In my case the answer to the tracking-requests took about 10 seconds to complete. The server had already recorded the events. But when the user navigated (or unloaded) the page while the request to server was still running, the stored events would not be marked as sent, since the http-request would not have had a response yet. 

For me this is resolved since I do not use the storage at the time.

Thanks! :-)

**Comment by eskalona** [comment]
Hi @tobias74, I have been trying with the latest version for the library (2.5.4), and still facing this issue if i open multiple tabs. The only way i can avoid duplicated events is by:
1. Setting "isStorageUseDisabled" to true.
2. Assigning a namePrefix (unique) for each new session (or tab) i open for the application.

When you referred to the new version (marking the events when the request is sent) which one are you talking about? 2.0.0? 

Thanks!!
